Kill eventdispatcher

nox opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15137)
<!-- Reviewable:end -->

nox committed:
Inline dom::eventdispatcher into dom::event

nox committed:
Kill AutoDOMEventMarker

highfive commented:
Heads up! This PR modifies the following files:
 * @fitzgen: components/script/dom/worker.rs, components/script/dom/document.rs, components/script/dom/eventtarget.rs, components/script/dom/htmlscriptelement.rs, components/script/dom/eventdispatcher.rs, components/script/dom/event.rs, components/script/dom/globalscope.rs
 * @kichjang: components/script/dom/worker.rs, components/script/dom/document.rs, components/script/dom/eventtarget.rs, components/script/dom/htmlscriptelement.rs, components/script/dom/eventdispatcher.rs, components/script/dom/event.rs, components/script/dom/globalscope.rs

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify script code, but no tests are modified. Please consider adding a test!

nox committed:
Remove some convenience methods on Event

They don't need to exist anymore now that dom::eventdispatcher has been removed.

nox commented:
r? @jdm

jdm commented:
@bors-servo: r+

bors-servo commented:
:pushpin: Commit 12d6d31 has been approved by `jdm`

<!-- @bors-servo r=jdm 12d6d31661867527430daff92778a7b4edcbdcb5 -->

bors-servo commented:
:hourglass: Testing commit 12d6d31 with merge 5c0c786...

bors-servo commented:
:broken_heart: Test failed - [linux-dev](http://build.servo.org/builders/linux-dev/builds/5314)

nox commented:
@bors-servo r=jdm

bors-servo commented:
:pushpin: Commit 3dd3736 has been approved by `jdm`

<!-- @bors-servo r=jdm 3dd3736b006218befb7c675bf3d797d6e9205439 -->

bors-servo commented:
:hourglass: Testing commit 3dd3736 with merge dc9706c...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/5000), [arm32](http://build.servo.org/builders/arm32/builds/5170), [arm64](http://build.servo.org/builders/arm64/builds/5214), [linux-dev](http://build.servo.org/builders/linux-dev/builds/5315), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/1989), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/1999), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/5285), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/5325), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/1987), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/2033), [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/453), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/456)
